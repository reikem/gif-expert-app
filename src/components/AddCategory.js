import React, { useState } from "react";

const AddCategory = ({ setCategory }) => {
  const [inputValue, setInputValue] = useState();

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    //console.log("Submit echo");
    if (inputValue.trim().length > 2) {
      setCategory((cats) => [...cats, inputValue]);
      setCategory("");
    }
  };

  return (
    <>
      <form onSubmit={handleSubmit}>
        <h3>{inputValue} </h3>
        <input type="text" value={inputValue} onChange={handleInputChange} />
      </form>
    </>
  );
};

export default AddCategory;
