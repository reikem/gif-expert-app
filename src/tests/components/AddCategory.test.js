import React from "react";
import { shallow } from "enzyme";
import "../../setupTests";
import AddCategory from "../../components/AddCategory";

describe("Pruebas de componente AddCategory", () => {
  const setCategory = jest.fn();
  let wrapper = shallow(<AddCategory setCategory={setCategory} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategory={setCategory} />);
  });

  test("Debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "hola mundo";
    input.simulate("change", { target: { value } });
    expect(wrapper.find("h3").text().trim()).toBe(value);
  });
  test("No debe de postear la informacion con submit", () => {
    wrapper.find("form").simulate("submit", {
      preventDefault() {},
    });
    expect(setCategory).not.toHaveBeenCalled();
  });
  test("debe llamar el setCategory y limpiar la caja de texto", () => {
    const value = "Hola mundo";
    //1.- simulamos el imputChange
    wrapper.find("input").simulate("change", { target: { value } });
    //2.- simula el submit
    wrapper.find("form").simulate("submit", {
      preventDefault() {},
    });
    //3.- setCategory se debe haber llamado
    expect(setCategory).toHaveBeenCalled();
    //muestra cuantas veces fue llamada la aplicacion
    expect(setCategory).toHaveBeenCalledTimes(1);
    //esperamos que la respuesta sea una funcion
    expect(setCategory).toHaveBeenCalledWith(expect.any(Function));
    //esperamos que despues de hacer la consulta el inputvalue tenga valor vacio
    expect(wrapper.find("input").prop("value")).toBe("");
  });
});
