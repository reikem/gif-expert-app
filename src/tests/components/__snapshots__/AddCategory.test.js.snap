// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de componente AddCategory Debe de mostrarse correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <h3>
       
    </h3>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
